syft/pkg/cataloger/dotnet/licenses.go

```diff
@@ -0,0 +1,608 @@
+}
+type nuspecDependencies struct {
+	Dependency []nuspecDependency `xml:"dependency"`
+}
+
+type nuspecMetaData struct { // MetaData
+	ID               string             `xml:"id"`
+	Version          string             `xml:"version"`
+	Title            string             `xml:"title,omitempty"`
+	Authors          string             `xml:"authors"`
+	Owners           string             `xml:"owners,omitempty"`
+	LicenseURL       string             `xml:"licenseUrl,omitempty"`
+	License          nuspecLicense      `xml:"license,omitempty"`
+	ProjectURL       string             `xml:"projectUrl,omitempty"`
+	IconURL          string             `xml:"iconUrl,omitempty"`
+	ReqLicenseAccept bool               `xml:"requireLicenseAcceptance"`
+	Description      string             `xml:"description"`
+	ReleaseNotes     string             `xml:"releaseNotes,omitempty"`
+	Copyright        string             `xml:"copyright,omitempty"`
+	Summary          string             `xml:"summary,omitempty"`
+	Language         string             `xml:"language,omitempty"`
+	Tags             string             `xml:"tags,omitempty"`
+	Dependencies     nuspecDependencies `xml:"dependencies,omitempty"`
+}
+
+type nuspecFiles struct {
+	File []nuspecFile `xml:"file"`
+}
+
+// nugetSpecification represents a .nuspec XML file found in the root of the .nupack or .nupkg files
+//
+// cf. https://learn.microsoft.com/en-us/nuget/reference/nuspec
+type nugetSpecification struct {
+	XMLName xml.Name       `xml:"package"`
+	Xmlns   string         `xml:"xmlns,attr,omitempty"`
+	Meta    nuspecMetaData `xml:"metadata"`
+	Files   nuspecFiles    `xml:"files,omitempty"`
+}
+
+// removeBOM removes any ByteOrderMark at the beginning of a given file content
+func removeBOM(input []byte) []byte {
+	if len(input) >= 4 {
+		if input[0] == 0 && input[1] == 0 && input[2] == 254 && input[3] == 255 {
+			// UTF-32 (BE)
+			return input[4:]
+		}
+		if input[0] == 255 && input[1] == 254 && input[2] == 0 && input[3] == 0 {
+			// UTF-32 (LE)
+			return input[4:]
+		}
+	}
+	if len(input) >= 3 {
+		if input[0] == 239 && input[1] == 187 && input[2] == 191 {
+			// UTF-8
+			return input[3:]
+		}
+	}
+	if len(input) >= 2 {
+		if input[0] == 254 && input[1] == 255 {
+			// UTF-16 (BE)
+			return input[2:]
+		}
+		if input[0] == 255 && input[1] == 254 {
+			// UTF-16 (LE)
+			return input[2:]
+		}
+	}
+	return input
+}
+
+type bytesReadCloser struct {
+	bytes.Buffer
+}
+
+func (brc *bytesReadCloser) Close() error {
+	return nil
+}
+
+func newBytesReadCloser(data []byte) *bytesReadCloser {
+	return &bytesReadCloser{
+		Buffer: *bytes.NewBuffer(data),
+	}
+}
+
+func extractLicensesFromNuGetContentFile(ctx context.Context, scanner licenses.Scanner, filePath string, nugetArchive *zip.Reader) []pkg.License {
+	out := []pkg.License{}
+
+	if nugetArchive != nil {
+		if licenseFile, err := nugetArchive.Open(filePath); err == nil {
+			licenseFileData, err := io.ReadAll(licenseFile)
+			licenseFile.Close()
```

Comment:
there is an `internal.CloseAndLogError(closer, someLocationString)` ([here](https://github.com/anchore/syft/blob/df3998b4f1c47f8d9fe8d08e244ed674eb38e9ef/internal/err_helper.go#L13)) that can be used which will automatically log on error; this could be useful in the event of read errors anywhere we are closing things -- most places would be something like this to be the most consistent:
```
defer internal.CloseAndLogError(licenseFile, filePath)
```